### PR TITLE
feat(NuGet): add support for specificying protocol version in registry URL

### LINKED ISSUE #5165: NuGet: Support for configuring feed version
**What would you like Renovate to be able to do?**
The protocol version of a NuGet feed is currently detected via the configured URL:

https://github.com/renovatebot/renovate/blob/d19709aebd9e82e2c8a978ec63d2c3baa7acca09/lib/datasource/nuget/index.ts#L9-L14

This works for the official feed but fails for self-hosted v3 feeds that don't end with `.json` (e.g. Artifactory).

**Describe the solution you'd like**
Configure the protocol version in the `nuget/registryUrls` setting:

```js
module.exports = {
  nuget: {
    registryUrls: [
      { value: "https://my-nuget-v3-feed", protocolVersion: "3" } 
    ],
```

**Describe alternatives you've considered**
- Not using v3 feeds
- Not using Artifactory

### PR DESCRIPTION
**Pull request by fgreinacher**
Closes #5165

### CODE REVIEW

**Comment by fgreinacher** [comment]
Note: aba6a53f687ca99ed938fdfa3f10e58df93de0ca adds configuration files for [VS Code Remote Development](https://code.visualstudio.com/docs/remote/remote-overview). This makes it super-easy to contribute to this repository via VS Code. If this should be a separate PR just say so :)

**Review by viceice — DISMISSED** [review]

**Comment by rarkins** [comment]
> Note: [aba6a53](https://github.com/renovatebot/renovate/commit/aba6a53f687ca99ed938fdfa3f10e58df93de0ca) adds configuration files for [VS Code Remote Development](https://code.visualstudio.com/docs/remote/remote-overview). This makes it super-easy to contribute to this repository via VS Code. If this should be a separate PR just say so :)

Yes, please make it a separate PR - sounds great

**Comment by fgreinacher** [comment]
> > Note: [aba6a53](https://github.com/renovatebot/renovate/commit/aba6a53f687ca99ed938fdfa3f10e58df93de0ca) adds configuration files for [VS Code Remote Development](https://code.visualstudio.com/docs/remote/remote-overview). This makes it super-easy to contribute to this repository via VS Code. If this should be a separate PR just say so :)
> 
> Yes, please make it a separate PR - sounds great

Done, see #5748

File: lib/datasource/nuget/index.ts [hunk 1]
```diff
@@ -6,17 +6,24 @@ import { GetReleasesConfig, ReleaseResult } from '../common';
 
 export { id } from './common';
 
-function detectFeedVersion(url: string): 2 | 3 | null {
+function parseRegistryUrl(
+  registryUrl: string
+): { feedUrl: string; protocolVersion: number } {
```
Comment (JamieMagee) [inline lib/datasource/nuget/index.ts]: A question, but not a request: Is it worth specifying the version as a type here. `protocolVersion: 2 | 3` as before, and `protocolVersionRegExp.exec(parsedUrl.hash) as 2 | 3`.

Comment (fgreinacher) [reply lib/datasource/nuget/index.ts]: I also thought about that, but decided against it for readability/simplicity.

**Pull request merged**